BlueprintUI/Sources/Layout/ConstrainedSizeFraction.swift

```diff
@@ -0,0 +1,88 @@
+import UIKit
+
+/// Constrains the size of the content element to a fraction of the parent's layout space.
+public struct ConstrainedSizeFraction: Element {
+    /// The element being constrained.
+    public var wrappedElement: Element
+    /// The fraction of the parent's horizontal layout space the content element should occupy. `nil` if the width is
+    /// unconstrained.
+    public var width: CGFloat?
+    /// The fraction of the parent's vertical layout space the content element should occupy. `nil` if the height is
+    /// unconstrained.
+    public var height: CGFloat?
+
+    /// Initializes with the given properties.
+    ///
+    /// - parameters:
+    ///   - width: The fraction of the parent's horizontal layout space the content element should occupy. `nil` if
+    ///   the width is unconstrained. By default, `nil`.
+    ///   - height: The fraction of the parent's vertical layout space the content element should occupy. `nil` if
+    ///   the height is unconstrained. By default, `nil`.
+    ///   - wrapping: The content element.
+    public init(width: CGFloat? = nil, height: CGFloat? = nil, wrapping wrappedElement: Element) {
+        precondition(width.map { $0 >= 0 && $0 <= 1} ?? true)
```

Comment:
Mind adding an error message as well? Eg `The provided fraction must be greater than zero or less than or equal to one, to be a valid fractional width.`

Reply:
(Also, the check should probably be `> 0`, not `>= 0`? It doesn't _hurt_, a zero-width item doesnt really mean anything.

Reply:
Updated, thanks.

Reply:
Sorry for the thrash here, but don't a lot of fraction based UI's end up using 0? A progress bar for example

Reply:
Seems like a valid use case to me. I'll restore it.

Reply:
Maybe in the case of zero, it makes sense to not include the child element from the elementContent and to short circuit the measurement? Eg still worried about all the possible divide by zeros we might run into in the wild, since elements rarely are given an empty size.

Reply:
Shouldn't we just fix those bugs if they come up? Elements aren't required not to have an empty size, are they? Math in Blueprint should defend against that. 

Aaaand one more thing I just thought of, why clamp at one? Making an element 2x the width of another one also seems like a perfectly valid use case 🤔

Reply:
But in this case, its the width of the _parent_, which is weird (and probably wrong, eg for views that clip their subviews).

Reply:
mmm good point.. would that make sense for `aligned` elements though? tbh I was thinking more about constraints, which you can set on any two elements in the tree 😅 leaving at 1 probably makes sense